frappe/core/doctype/role_profile/role_profile.py

```diff
@@ -41,9 +41,15 @@ def on_update(self):
 		for user, role in all_current_roles:
 			user_roles[user].add(role)
 
-		role_profile_roles = {role.role for role in self.roles}
-		for user, roles in user_roles.items():
-			if roles != role_profile_roles:
-				user = frappe.get_doc("User", user)
-				user.roles = []
-				user.add_roles(*role_profile_roles)
+		if len(user_roles) > 10:
+			frappe.enqueue("frappe.core.doctype.role_profile.role_profile.update_roles", doc=self, user_roles=user_roles)
+		else:
+			update_roles(self, user_roles)
+
+def update_roles(doc, user_roles):
```

Comment:
This is a very weird splitting of code. Why not enqueue the entire method instead of splitting the computation of user_roles?

Reply:
ok, maybe!